### DISCUSSION TITLE
Validation error: The character "-" is not allowed in field names. How to allow "-" in the scheme

### DISCUSSION THREAD

**Discussion by deback** [prisma/prisma#20423] [Q&A] [answered]
i am trying to combine keycloak provider with the the prisma adapter.
I get the error "Unknown argument `not-before-policy`. Available options are listed in green.

When I include "not-before-policy" in my Account schema, I get the error:
"Validation error: The character '-' is not allowed in field names."

Is it possible to allow '-' in the scheme or is there another solution?

**Comment by nurul3101** [comment]
Hello @deback 👋 

Prisma schema does not support '-' in the field names. This is because Prisma generates a TypeScript client based on your schema, and in TypeScript, '-' is not allowed in variable or property names because it's an operator used for subtraction.

You can use the `@map` directive ([reference](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#map)) to map a database column that includes a '-' to a different field name in your Prisma model.

Your Account model could look something like:
```prisma
model Account {
  notBeforePolicy Int @map("not-before-policy")
 // other fields
}
```

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by deback** [reply]
<img width="897" alt="image" src="https://github.com/prisma/prisma/assets/301778/ba676b22-f5e8-458e-b028-713315a6525b">

**Reply by deback** [reply]
I tried: 
`not_before_policy        Int?    @map("not-before-policy")`
unfortunately this does not work, "not-before-policy" is what comes from keycloak.

**Reply by deback** [reply, marked as answer]
my solution now is to use the mongodb adapter instead of the prisma adapter.

**Reply by nmcc1212** [reply]
Running in to the same issue, unfortunately I'm using Postgres not Mongo
